setup.py

```diff
@@ -5,68 +5,77 @@
     find_packages,
 )
 
-
-deps = {
-    'test': [
-        "pytest>=6.2.5,<8",
-        "flaky>=3.2.0,<4",
-        "pluggy>=0.7.1,<1",
+extras_require = {
+    "test": [
+        "pytest>=7.0.0",
+        "pytest-xdist>=2.4.0",
+        "flaky>=3.2.0",
     ],
-    'lint': [
-        "flake8>=3.9.2,<4",
+    "lint": [
+        "flake8==5.0.4",  # flake8 claims semver but adds new warnings at minor releases, leave it pinned.
+        "flake8-bugbear==23.3.12",  # flake8-bugbear does not follow semver, leave it pinned.
+        "isort>=5.10.1",
+        "black>=23",
         "importlib-metadata<5;python_version<'3.8'",
     ],
-    'dev': [
-        "bumpversion>=0.5.3,<1",
+    "docs": [
+        "towncrier>=21,<22",
+    ],
+    "dev": [
+        "bumpversion>=0.5.3",
+        "pytest-watch>=4.1.0",
+        "tox>=3.28.0",
```

Comment:
I think we may need to use tox>=4 since we've converted to `allowlist`?

Reply:
Update: nope, all the py37-install-geth tests fail with tox 4.